classes/core/traits/EntityUpdate.php

```diff
@@ -74,7 +74,15 @@ public function updateSettings(array $props, int $modelId, $schema = null): void
                 $deleteSettings[] = $propName;
                 continue;
             }
+
             if (!empty($propSchema->multilingual)) {
+                // first we will delete the settings entries for the locale keys which are not present in update query
+                DB::table($this->getSettingsTable())
+                    ->where($this->getPrimaryKeyName(), '=', $modelId)
+                    ->where('setting_name', '=', $propName)
+                    ->whereNotIn('locale', array_keys($props[$propName]))
+                    ->delete();
+
```

Comment:
@Vitaliy-1 can you please take a look at this modification .

Reply:
> Do you mean that without this modification the locale value can't be updated to null in usual way? E.g., if we updating the description of the Announcement through the form by removing one locale data, it won't be removed from the database?

that's not the case . it works that way properly . 

> I'm thinking about situation when locale has been turned of globally. This means that values for the old locale will be deleted. And this removal will be inconsistent, per update request for each entity. But I prefer to preserve old locale values in case they would be needed again - when the user turns on the old locale again

considering this case (which I did not consider), we should not remove data for every other local . I am reverting this change .